cvmfs/options.cc

```diff
@@ -311,14 +320,34 @@ void OptionsManager::PopulateParameter(
              param.c_str(), iter->second.c_str(), val.value.c_str());
     return;
   }
-
+  ParseValue(&val, opt_templ_mgr);
   config_[param] = val;
   if (taint_environment_) {
     int retval = setenv(param.c_str(), val.value.c_str(), 1);
     assert(retval == 0);
   }
 }
 
+void OptionsManager::ParseValue(ConfigValue *val,
+  OptionsTemplatingManager *opt_templ_mgr) {
+  if (opt_templ_mgr == NULL) {
+    // No template manager -> nothing to do here
+    return;
+  }
+  vector<string> tokens = SplitString(val->value, '@');
```

Comment:
Apologies: I forgot that we already have a `ReplaceAll` utility function in util/string.cc. So you can simply iterate through all available template identifier and replace them.  In fact, this could be done in the `OptionsTemplateManager` that can iterate over all the templates.

Reply:
In this case I would not be able to detect undefined template variables though (previous note). Depends on what is more important...
Alternatively one could use the simple finite state machine system that You had already mentioned (probably best implemented in the `OptionsTemplateManager` then).